File: codeforces/23.beautiful_year/beautiful_year.py
# Author: José Rodolfo (jric2002)
year = int(input())
is_different = False
while (is_different != True):
  year = year + 1
  numbers = str(year)
  repeated_digits = 0
  for position in range(0, len(numbers)):
    number = numbers[position]
    for i in range(position + 1, len(numbers)):
      if (number == numbers[i]):
        repeated_digits += 1
  if (repeated_digits == 0):
    is_different = True
print(year)

File: codeforces/23.beautiful_year_[271A]/beautiful_year.cpp
#include <iostream>
#include <cstring>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int quantity_digits = 4;
  int year;
  string numbers;
  char number;
  bool is_different = false;
  int repeated_digits;
  cin >> year;
  while (!is_different) {
    year = year + 1;
    numbers = to_string(year);
    repeated_digits = 0;
    for (unsigned short int position = 0; position < quantity_digits; position++) {
      number = numbers[position];
      for (unsigned short int i = position + 1; i < quantity_digits; i++) {
        if (number == numbers[i]) {
          repeated_digits++;
        }
      }
    }
    if (repeated_digits == 0) {
      is_different = true;
    }
  }
  cout << year << endl;
  return 0;
}